Scripts/General/ModMercs.lua

```diff
@@ -0,0 +1,732 @@
+            RemoveMonster(mon)
+        end
+    end
+
+    -- Reset
+    Timer(
+        function()
+            local UpgradeUseLevel   = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Charges) or 1
+            local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+            MercSaveData.FightsLeft = Merc.FightsMax[UpgradeUseLevel+1] or 1
+        end, 
+        const.Day, const.Hour, true)
+
+    ExitScreen()
+end
+
+function Merc_IsSpecial(Merc, t)
+    
+    if Merc.Ability == nil              then return false end
+    if type(Merc.Ability) ~= "string"   then return false end
+    if string.len(Merc.Ability) == 0    then return false end
+    return true
+end
+
+function Merc_IsSpecialTag(Merc, tag)
+
+    if not Merc_IsSpecial(Merc, nil) then return false end
+
+    for word in string.gmatch(tag, "([^,]+)") do
+        if word == tag then
+            return true
+        end
+    end
+
+    return false
+end
+
+function Merc_ShowMessageAboutAbility(Merc, t) 
+
+    if Merc_IsSpecial(Merc, t) then
+        Message(Merc.Credentials.TextSpecial)
+    end
+end
+
+function Merc_ConsumeCharge(Merc)
+
+    local MercSaveData = Merc_GetSaveDataByID(Merc.NPC_ID)
+    MercSaveData.FightsLeft = math.max(0, MercSaveData.FightsLeft - 1)
+end
+
+function Merc_ParseDamageString(damageString)
+
+    local count, sides, add = damageString:match("(%d+)D(%d+)%+?(%d*)")
+
+    count   = tonumber(count)
+    sides   = tonumber(sides)
+    add     = (add ~= "" and tonumber(add)) or 0
+
+    local Attack1 = {
+        DamageAdd       = add,
+        DamageDiceCount = count,
+        DamageDiceSides = sides,
+    }
+
+    return Attack1
+end
+
+-- @todo Rework/Finish
+function Merc_ShowInfo(Merc)
+
+    local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+    
+    local UpgMax            = const.Mercenary.UpgradeLimit
+    local UpgradeHPLevel    = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.HP)
+    local UpgradeACLevel    = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.AC)
+    local UpgradeMonLevel   = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Level)
+    local UpgradeDmg1Level  = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.MeleeDamage)
+    local UpgradeDmg2Level  = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.RangeDamage)
+    local UpgradeSumLevel   = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Charges)
+
+    local hp                = Merc.FullHP[UpgradeHPLevel+1]       or Merc.FullHP[1]
+    local ac                = Merc.AC[UpgradeACLevel+1]           or Merc.AC[1]
+    local level             = Merc.Level[UpgradeMonLevel+1]       or Merc.Level[1]
+    local attack1           = Merc.Attack1[UpgradeDmg1Level+1]    or Merc.Attack1[1]
+    local attack2           = Merc.Attack2[UpgradeDmg2Level+1]    or Merc.Attack2[1]
+    local fightsLeft        = Merc.FightsMax[UpgradeSumLevel+1]   or Merc.FightsMax[1]
+
+    local meleeStr          = ""
+    local meleeUpdStr       = ""
+    if (attack1 and string.len(attack1) > 0) then
```

Comment:
Hmm, maybe this may be worth wrapping? Maybe attack1 should not be type of string but sth parsed at the moment of assigning it to the `attack` local variable?
I can imagine the situation where developer may forget to check `string.len(...) > 0.